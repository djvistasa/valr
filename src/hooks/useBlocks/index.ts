import { useCallback, useState } from 'react';
import { BlocksApi } from '../../api';
import { IBlock } from '../../interfaces';

function useBlocks() {
  const [blocks, setBlocks] = useState<IBlock[]>([]);
  const [block, setBlock] = useState<IBlock>();

  const getBlocksByAsset = useCallback(async (asset: string) => {
    const { error, ok, result } = await BlocksApi.getBlocksByAsset(asset);

    if (error) {
      return alert(error);
    }

    if (ok && result && result.pageProps && result.pageProps.latestBlocks) {
      setBlocks(result.pageProps.latestBlocks);
    }
  }, []);

  const getBlockByAssetAndId = useCallback(
    async (asset: string, id: string) => {
      const { error, ok, result } = await BlocksApi.getBlockByAssetAndId(
        id,
        asset,
      );

      if (error) {
        return alert(error);
      }

      if (ok && result) {
        setBlock(result);
      }
    },
    [],
  );

  return {
    getBlocksByAsset,
    getBlockByAssetAndId,
    block,
    blocks,
  };
}

export default useBlocks;
