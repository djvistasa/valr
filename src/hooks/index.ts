import useBlocks from './useBlocks';

export { useBlocks };
