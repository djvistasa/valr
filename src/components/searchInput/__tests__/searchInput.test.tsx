import React from 'react';
import { test, expect, vi } from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import { useTestComponentWithTheme } from '../../../hooks';
import SearchInput from '..';

test('does searchInput render', () => {
  const mountWithTheme = useTestComponentWithTheme();
  const { getByTestId } = render(
    mountWithTheme(<SearchInput onSearch={() => {}} testID="search-input" />)
  );

  const input = getByTestId('search-input');

  expect(input).toMatchSnapshot();
});

test('does onChange get called', () => {
  const onChange = vi.fn();
  const mountWithTheme = useTestComponentWithTheme();
  const { getByTestId } = render(
    mountWithTheme(<SearchInput onSearch={onChange} testID="search-input" />)
  );

  const input = getByTestId('search-input');

  fireEvent.change(input, { target: { value: 'South Africa' } });

  expect(onChange).toHaveBeenCalledTimes(1);

  expect(input.value).toBe('South Africa');
});
