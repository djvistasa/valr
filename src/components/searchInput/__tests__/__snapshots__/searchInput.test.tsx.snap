// Vitest Snapshot v1

exports[`does searchInput render 1`] = `
<input
  class="sc-gswNZR bnLElW"
  data-testid="search-input"
  placeholder="Search for a country..."
/>
`;
